pkg/api/component_report.go

```diff
@@ -64,6 +74,19 @@ const (
 		SELECT * FROM deduped_testcases WHERE row_num = 1`
 )
 
+type GeneratorType string
+
+var (
+	ComponentReport        GeneratorType = "ComponentReport"
+	SampleTestStatus       GeneratorType = "SampleTestStatus"
+	BaseTestStatus         GeneratorType = "BaseTestStatus"
+	SampleJobRunTestStatus GeneratorType = "SampleJobRunTestStatus"
+	BaseJobRunTestStatus   GeneratorType = "BaseJobRunTestStatus"
+	TriagedIncidents       GeneratorType = "TriagedIncidents"
+	TestDetailsReport      GeneratorType = "TestDetailsReport"
+	TestVariants           GeneratorType = "TestVariants"
```

Comment:
I'm not a huge fan of this.  This was the thing I mentioned I was thinking about.  I can see why the GeneratorType is useful to store, but it's really more a product of some of the downsides of the current implementation.

I wish each of the generators was a different type instead of overloading componentReportGenerator to do many different things. Then each one could implement a new Generator interface, which could simplify the caching layer (anything implementing Generator could be cached, without having to pass so much to  getDataFromCacheOrGenerate)

And that would facilitate getting the generators out of the API layer entirely.  We've coupled the API and data model layers a bit too tightly IMHO.

Reply:
I'm open to working through this with you either in this PR or a latter one.  I have started to create new generators for the base / sample query and triaged incidents.  Would you be open to the Generator having an interface that provided the cache key which it could optionally add a prefix to guarantee uniqueness for it's generator type?

Reply:
> Would you be open to the Generator having an interface that provided the cache key which it could optionally add a prefix to guarantee uniqueness for it's generator type?

Yea, exactly - I was thinking the interface impl would manage it's cache key.